compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -461,25 +453,26 @@ private fun ContributedBinding.toGeneratedMethod(
 private val Collection<GeneratedMethod>.specs get() = map { it.spec }
 
 private class MergedClassKey(
-  val scope: FqName,
-  val clazz: KtClassOrObject
+  val scope: ClassReference,
+  val clazz: ClassReference
 ) {
-  private val classFqName = clazz.requireFqName()
+  private val scopeFqName = scope.fqName
+  private val classFqName = clazz.fqName
 
   override fun equals(other: Any?): Boolean {
     if (this === other) return true
     if (javaClass != other?.javaClass) return false
 
     other as MergedClassKey
 
-    if (scope != other.scope) return false
+    if (scopeFqName != other.scopeFqName) return false
     if (classFqName != other.classFqName) return false
 
     return true
   }
 
   override fun hashCode(): Int {
-    var result = scope.hashCode()
+    var result = scopeFqName.hashCode()
     result = 31 * result + classFqName.hashCode()
     return result
   }
```

Comment:
These functions are redundant now. You can make `MergedClassKey` a data class, because `ClassReference` itself implements equals() and hashcode(). That wasn't possible before, because `KtClassOrObject` didn't implement these methods.